src/interpreter.js

```diff
@@ -40,26 +40,27 @@ import {
 // I know, I know... But this is not meant to be nice or final code. Just a quick rough approach to get a better feeling on the AST shape.
 
 const compileSentence = (sentence) => {
-  if(sentence.nodeType === 'VariableDeclaration') {
-    const { variable, writeable, value } = sentence
-    return ` ${writeable ? 'let' : 'const'} ${variable.name} = ${compileExpression(value)} ;`
-  }
-  if(sentence.nodeType === 'Assignment') {
-    const { variable, value } = sentence
-    return ` ${variable.name} = ${compileExpression(value)} ;`
+  switch (sentence.nodeType) {
```

Comment:
Beware of switch... It shares the same context between all cases, so you can't extract variables like this:

switch(node.type){
  case 'X': const ({a, b}) = node ...
  case 'Y': const ({a, b}) = node ... // a and b are already defined :(
}

I didn't know you could do a case with braces instead of break, but it may still be error prone for future contributors.

Anyway, this should not be a problem if we go for the implementation discussed ahead